fcos-graph-builder/src/main.rs

```diff
@@ -144,7 +144,10 @@ pub(crate) async fn gb_serve_graph(
 
     let cached_graph = addr.send(scraper::GetCachedGraph { stream }).await??;
 
-    let arch_graph = policy::pick_basearch(cached_graph, basearch)?;
+    let arch_graph = match policy::pick_basearch(cached_graph, basearch) {
+        Err(err) => return Ok(HttpResponse::BadRequest().body(err.to_string())),
```

Comment:
There is still a bit of groundwork missing on the topic of returning error bodies, see https://github.com/coreos/zincati/blob/master/docs/development/cincinnati/protocol.md#errors.

Can you please just return an empty BadRequest to the client, and `log::error!()` the failure message locally?

Reply:
Done.
Not sure returning an empty BadRequest is a good idea if the zincati clients expects a json body. It's probably fine if zincati is robust enough.

Reply:
Zincati is fine with empty errors. For reference, the plaintext error would have been fine too, but it would have been discarded as not matching the expected JSON format, so it was mostly pointless.